README.md

```diff
@@ -8,223 +8,223 @@ extensions:
+
+// register Google endpoint using
+authenticator.endpoints.registerGoogleAuth('client id here');
+
+// register Facebook endpoint using
+authenticator.endpoints.registerFacebookAuth('client id here');
+
+// register any 3rd-Party OAuth Implicit Provider using
+authenticator.endpoints.add('Name of provider', { /* Endpoint Configuration - see office-js-helpers/src/authentication/endpoint.manager.ts */ })
+
+// register Microsoft endpoint by overriding default values
+authenticator.endpoints.registerMicrosoftAuth('client id here', {
+    redirectUrl: 'redirect url here',
+    scope: 'list of valid scopes here'
+});
+```
+
+#### Authentication
+To authenticate against the registered endpoint, do one of the following:
+
+```javascript
+authenticator
+    .authenticate('name of endpoint')
+    .then(function(token) { /* handle success here */ })
+    .catch(OfficeHelpers.Utilities.log);
+
+// for the default Microsoft endpoint
+authenticator
+    .authenticate(OfficeHelpers.DefaultEndpoints.Microsoft)
+    .then(function (token) { /* Microsoft Token */ })
+    .catch(OfficeHelpers.Utilities.log);
+
+// for the default AzureAD endpoint
+authenticator
+    .authenticate(OfficeHelpers.DefaultEndpoints.AzureAD)
+    .then(function (token) { /* Microsoft Token */ })
+    .catch(OfficeHelpers.Utilities.log);
+
+// for the default Google endpoint
+authenticator
+    .authenticate(OfficeHelpers.DefaultEndpoints.Google)
+    .then(function (token) { /* Google Token */ })
+    .catch(OfficeHelpers.Utilities.log);
+
+// for the default Facebook endpoint
+authenticator
+    .authenticate(OfficeHelpers.DefaultEndpoints.Facebook)
+    .then(function (token) { /* Facebook Token */ })
+    .catch(OfficeHelpers.Utilities.log);
+```
+If the user rejects the grant to the application then you will receive an error in the `catch` function.
+
+#### Getting a cached token
+By default the tokens are cached to the LocalStorage and upon expiry the AuthDialog is invoked again. You can also pass the `force` parameter as `true` as the second input to `authenticator.authenticate()` to re-authenticate the user.
+
+```javascript
+authenticator
+    .authenticate('name of endpoint')
+    .then(function(token) {
+    /*
+        `token` is either cached or newly obtained upon expiry.
+    */
+    })
+    .catch(OfficeHelpers.Utilities.log);
+
+authenticator
+    .authenticate('name of endpoint', true /* force re-authentication */)
+    .then(function(token) {
+    /*
+        `token` is newly obtained.
+    */
+    })
+    .catch(OfficeHelpers.Utilities.log);
+
+// get the cached token if any. returns null otherwise.
+var token = authenticator.tokens.get('name of endpoint');
+```
+If a cached token expires, then the dialog is automatically launched to re-authenticate the user.
+> Note on Refresh Tokens: By default, Implicit OAuth does not support Token Refresh as a security measure. This is because Access Tokens cannot be securely stored inside of a JavaScript client.

```

Comment:
```suggestion
> **Note on Refresh Tokens**: By default, Implicit OAuth does not support Token Refresh as a security measure. This is because Access Tokens cannot be securely stored inside a JavaScript client.
```